File: module_main/src/main/java/com/fly/tour/main/MainActivity.java
package com.fly.tour.main;

import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.os.Bundle;
import android.view.MenuItem;

import com.alibaba.android.arouter.facade.annotation.Autowired;
import com.alibaba.android.arouter.launcher.ARouter;
import com.fly.tour.common.base.BaseActivity;
import com.fly.tour.common.manager.NewsDBManager;
import com.fly.tour.common.provider.IFindProvider;
import com.fly.tour.common.provider.IMeProvider;
import com.fly.tour.common.provider.INewsProvider;
import com.fly.tour.main.entity.MainChannel;
import com.trello.rxlifecycle2.components.support.RxAppCompatActivity;

/**
 * Description: <BaseRefreshLayout><br>
 * Author:      mxdl<br>
 * Date:        2019/02/25<br>
 * Version:     V1.0.0<br>
 * Update:     <br>
 */
public class MainActivity extends BaseActivity {
    @Autowired(name = "/news/main")
    INewsProvider mNewsProvider;

    @Autowired(name = "/find/main")
    IFindProvider mFindProvider;

    @Autowired(name = "/me/main")
    IMeProvider mMeProvider;

    private Fragment mNewsFragment;
    private Fragment mFindFragment;
    private Fragment mMeFragment;
    private Fragment mCurrFragment;

    @Override
    public int onBindLayout() {
        return R.layout.activity_main;
    }

    @Override
    public void initView() {

        BottomNavigationView navigation = findViewById(R.id.navigation);
        navigation.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                int i = menuItem.getItemId();
                if (i == R.id.navigation_trip) {
                    switchContent(mCurrFragment, mNewsFragment, MainChannel.NEWS.name);
                    mCurrFragment = mNewsFragment;

                    return true;
                } else if (i == R.id.navigation_discover) {
                    switchContent(mCurrFragment, mFindFragment, MainChannel.FIND.name);
                    mCurrFragment = mFindFragment;

                    return true;
                } else if (i == R.id.navigation_me) {
                    switchContent(mCurrFragment, mMeFragment, MainChannel.ME.name);
                    mCurrFragment = mMeFragment;

                    return true;
                }
                return false;
            }
        });
        if(mNewsProvider != null){
            mNewsFragment = mNewsProvider.getMainNewsFragment();
        }
        if(mFindProvider != null){
            mFindFragment = mFindProvider.getMainFindFragment();
        }
        if(mMeProvider != null){
            mMeFragment = mMeProvider.getMainMeFragment();
        }
        mCurrFragment = mNewsFragment;
        if(mNewsFragment != null){
            getSupportFragmentManager().beginTransaction().replace(R.id.frame_content, mNewsFragment, MainChannel.NEWS.name).commit();
        }
    }

    @Override
    public void initData() {

    }

    public void switchContent(Fragment from, Fragment to, String tag) {
        if (from == null || to == null) {
            return;
        }
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        if (!to.isAdded()) {
            transaction.hide(from).add(R.id.frame_content, to, tag).commit();
        } else {
            transaction.hide(from).show(to).commit();
        }
    }
}

File: module_main/src/main/java/com/fly/tour/main/MainActivity.kt
package com.fly.tour.main

import androidx.fragment.app.Fragment
import com.alibaba.android.arouter.facade.annotation.Autowired
import com.fly.tour.common.base.BaseActivity
import com.fly.tour.common.provider.IFindProvider
import com.fly.tour.common.provider.IMeProvider
import com.fly.tour.common.provider.INewsProvider
import com.fly.tour.common.util.ToastUtil
import com.fly.tour.main.entity.MainChannel
import kotlinx.android.synthetic.main.activity_main_index.*

class MainActivity : BaseActivity() {
    @Autowired(name = "/news/main")
    @JvmField var mNewsProvider: INewsProvider? = null

    @Autowired(name = "/find/main")
    @JvmField var mFindProvider: IFindProvider? = null

    @Autowired(name = "/me/main")
    @JvmField var mMeProvider: IMeProvider? = null

    private var mNewsFragment: Fragment? = null
    private var mFindFragment: Fragment? = null
    private var mMeFragment: Fragment? = null
    private var mCurrFragment: Fragment? = null

    override fun onBindLayout(): Int {
        return R.layout.activity_main_index
    }

    override fun initListener() {
        navigation.setOnNavigationItemSelectedListener {
            when (it.itemId) {
                R.id.navigation_news -> {
                    switchContent(mCurrFragment, mNewsFragment, MainChannel.NEWS.name)
                    mCurrFragment = mNewsFragment
                }
                R.id.navigation_discover -> {
                    switchContent(mCurrFragment, mFindFragment, MainChannel.FIND.name)
                    mCurrFragment = mFindFragment
                }
                R.id.navigation_me -> {
                    switchContent(mCurrFragment, mMeFragment, MainChannel.ME.name)
                    mCurrFragment = mMeFragment
                }
            }
            return@setOnNavigationItemSelectedListener true
        }

    }
    override fun initData() {
        mNewsFragment = mNewsProvider?.mainNewsFragment
        mFindFragment = mFindProvider?.mainFindFragment
        mMeFragment = mMeProvider?.mainMeFragment

        mCurrFragment = mNewsFragment
        if (mNewsFragment != null) {
            supportFragmentManager.beginTransaction()
                .replace(R.id.frame_content, mNewsFragment as Fragment, MainChannel.NEWS.name)
                .commit()
        }
    }

    fun switchContent(from: Fragment?, to: Fragment?, tag: String) {
        if (from == null || to == null) {
            return
        }
        val beginTransaction = supportFragmentManager.beginTransaction()
        if (!to.isAdded) {
            beginTransaction.hide(from).add(R.id.frame_content, to, tag).commit()
        } else {
            beginTransaction.hide(from).show(to).commit()
        }
    }
}
